milli/src/update/index_documents/helpers/merge_functions.rs

```diff
@@ -44,6 +45,27 @@ pub fn merge_roaring_bitmaps<'a>(_key: &[u8], values: &[Cow<'a, [u8]>]) -> Resul
     }
 }
 
+pub fn merge_btreeset_string<'a>(_key: &[u8], values: &[Cow<'a, [u8]>]) -> Result<Cow<'a, [u8]>> {
+    if values.len() == 1 {
+        Ok(values[0].clone())
+    } else {
+        // TODO improve the perf by using a `#[borrow] Cow<str>`.
+        let strings: BTreeSet<String> = values
+            .iter()
+            .map(AsRef::as_ref)
+            .map(serde_json::from_slice::<BTreeSet<String>>)
+            .map(StdResult::unwrap)
+            .reduce(|mut current, new| {
+                for x in new {
+                    current.insert(x);
+                }
+                current
```

Comment:
What do you think of [using `union`](https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html#method.union) instead of using a for `loop`?

It's not as practical as `RoaringBitmap` unfortunately.

Reply:
I will not use the `BitOr` trait as, unfortunately, it will clone the values. Here, we move the values from one set to the other without duplication.